homeassistant/components/zigate.py

```diff
@@ -0,0 +1,278 @@
+        entity = hass.data[DATA_ZIGATE_ATTRS].get(key)
+        if entity:
+            if entity.hass:
+                entity.schedule_update_ha_state()
+        key = '{}-{}-{}'.format(device.addr,
+                                'light',
+                                attribute['endpoint'],
+                                )
+        entity = hass.data[DATA_ZIGATE_ATTRS].get(key)
+        if entity:
+            if entity.hass:
+                entity.schedule_update_ha_state()
+        entity = hass.data[DATA_ZIGATE_DEVICES].get(device.addr)
+        if entity:
+            if entity.hass:
+                entity.schedule_update_ha_state()
+
+    zigate.dispatcher.connect(attribute_updated,
+                              zigate.ZIGATE_ATTRIBUTE_UPDATED, weak=False)
+
+    def device_updated(**kwargs):
+        device = kwargs['device']
+        _LOGGER.debug('Update device {}'.format(device))
+        entity = hass.data[DATA_ZIGATE_DEVICES].get(device.addr)
+        if entity:
+            if entity.hass:
+                entity.schedule_update_ha_state()
+        else:
+            _LOGGER.debug('Device not found {}, adding it'.format(device))
+            device_added(device=device)
+
+        zigate.dispatcher.connect(device_updated,
+                                  zigate.ZIGATE_DEVICE_UPDATED, weak=False)
+        zigate.dispatcher.connect(device_updated,
+                                  zigate.ZIGATE_ATTRIBUTE_ADDED, weak=False)
+
+    def zigate_reset(service):
+        z.reset()
+
+    def permit_join(service):
+        z.permit_join()
+
+    def zigate_cleanup(service):
+        '''
+        Remove missing device
+        '''
+        z.cleanup_devices()
+
+    def start_zigate(service_event):
+        z.autoStart()
+        z.start_auto_save()
+        # firt load
+        for device in z.devices:
+            device_added(device=device)
+
+        load_platform(hass, 'sensor', DOMAIN, {}, config)
+        load_platform(hass, 'binary_sensor', DOMAIN, {}, config)
+        load_platform(hass, 'switch', DOMAIN, {}, config)
+        load_platform(hass, 'light', DOMAIN, {}, config)
+
+    def stop_zigate(service_event):
+        z.save_state()
+        z.close()
+
+    def refresh_device(service):
+        addr = service.data.get(ADDR)
+        if addr:
+            z.refresh_device(addr)
+        else:
+            for device in z.devices:
+                device.refresh_device()
+
+    def network_scan(service):
+        z.start_network_scan()
+
+    def raw_command(service):
+        cmd = service.data.get('cmd')
+        data = service.data.get('data', '')
+        z.send_data(cmd, data)
+
+    def identify_device(service):
+        addr = service.data.get('addr')
+        z.identify_device(addr)
+
+    def initiate_touchlink(service):
+        z.initiate_touchlink()
+
+    def touchlink_factory_reset(service):
+        z.touchlink_factory_reset()
+
+    hass.bus.listen_once(EVENT_HOMEASSISTANT_START, start_zigate)
```

Comment:
Why do we need to wait with starting zigate and loading the platforms until home assistant has started?